src/main/java/at/hannibal2/skyhanni/features/fishing/TotemOfCorruption.kt

```diff
@@ -0,0 +1,155 @@
+package at.hannibal2.skyhanni.features.fishing
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.ReceiveParticleEvent
+import at.hannibal2.skyhanni.utils.EntityUtils
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.sendTitle
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.SoundUtils.playPlingSound
+import at.hannibal2.skyhanni.utils.SpecialColour
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.TimeUnit
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.getLorenzVec
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.entity.item.EntityArmorStand
+import net.minecraft.util.EnumParticleTypes
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+private val config get() = SkyHanniMod.feature.fishing.totemOfCorruption
+
+private var display = emptyList<String>()
+private var totems: List<Totem> = emptyList()
+
+class TotemOfCorruption {
+
+    private val group = RepoPattern.group("features.fishing.totemofcorruption")
+    private val totemNamePattern by group.pattern(
+        "totemname",
+        "§5§lTotem of Corruption"
+    )
+    private val timeRemainingPattern by group.pattern(
+        "timeremaining",
+        "§7Remaining: §e(?:(?<min>\\d+)m )?(?<sec>\\d+)s"
+    )
+    private val ownerPattern by group.pattern(
+        "owner",
+        "§7Owner: §e(?<owner>.+)"
+    )
+
+    @SubscribeEvent
+    fun onRender(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isOverlayEnabled() || display.isEmpty()) return
+        config.position.renderStrings(display, posLabel = "Totem of Corruption")
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!event.repeatSeconds(2)) return
+        if (!isOverlayEnabled()) return
+
+        totems = getTotems()
+            .filterNotNull()
+            .mapNotNull { totem ->
+                val timeRemaining = getTimeRemaining(totem)
+                val owner = getOwner(totem)
+                if (timeRemaining != null && owner != null) {
```

Comment:
create an extra function for "is this a real totem" check. then you can use gettime remaining ?: return null and get owner ?: return null